### ISSUE TITLE
NGINX byte-range support is not configurable

### ISSUE THREAD

**Issue by pcm32** [galaxyproject/ansible-galaxy-extras#226]
Javascript white listed tools (with client side only viz) and some interactive environments might benefit from byte-range support through NGINX for some requests (I'm working on one of these that requires it). This is currently not supported for configuration, which is done through http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_force_ranges

This is turned off by default.

If there is no opposition, I would like to add an optional setup for this

**Comment by abretaud** [comment]
:+1: the jbrowse tool needs this too

**Comment by pcm32** [comment]
My question would be then, the optional `proxy_force_ranges on` should go here https://github.com/galaxyproject/ansible-galaxy-extras/blob/master/templates/nginx.conf.j2#L126 ?

or somewhere else?

**Comment by mvdbeek** [comment]
Hmm, that doesn't sound completely right. The primary target for byte-range requests is the datasets controller, and this doesn't need the the `proxy_force_ranges ` flag on our setup, which uses ansible-galaxy-extras to configure nginx. Do you have a specific route / test command that is failing ?
Alternatively here's a command to verify it works without this flag:
```
curl -H "range: bytes=0-8" https://usegalaxy.org/datasets/bbd44e69cb8906b5feba48dc6b55f34f/display
```
usegalaxy.org isn't using this flag either as far as I can tell: https://github.com/galaxyproject/usegalaxy-playbook/blob/master/env/main/templates/nginx/usegalaxy.j2

**Comment by pcm32** [comment]
Yes, requests would be inside the `/datasets/<uuid>/` as you suggest.
